### PR TITLE
Added support for pgBouncer v1.23

### PR DESCRIPTION
**Pull request by rahulreddy15**
Testcases added for v1.23
Docker compose file added for setting up pgBouncer environment in the future.

### CODE REVIEW

File: src/metrics/metrics_test.go [hunk 1]
```diff
@@ -594,8 +675,14 @@ func TestPopulatePgBouncerMetrics(t *testing.T) {
...
 
-			mock.ExpectQuery("SHOW STATS;").
-				WillReturnRows(pgbouncerStatsRows)
+			if testCase.pgbouncerStatsRows == nil {
+				mock.ExpectQuery("SHOW STATS;").
+					WillReturnRows(pgbouncerStatsRows)
+			} else {
+				mock.ExpectQuery("SHOW STATS;").
+					WillReturnRows(testCase.pgbouncerStatsRows)
+			}
```
Comment (sigilioso) [inline src/metrics/metrics_test.go:684]: Nit: we could update the previous test cases with:

```go
// ...
pgbouncerStatsRows: pgbouncerStatsRows,
// ...
```

That way we wouldn't need this if/else clasue, and the results would be explicit for each test case. We could even improve the `pgbouncerStatsRows` global variable name to say that these are the expected rows for versions prior to PgBouncer 23. Example: `pgbouncerPriorTo23StatsRows`.

File: src/metrics/metrics_test.go [hunk 2]
```diff
@@ -606,7 +693,13 @@ func TestPopulatePgBouncerMetrics(t *testing.T) {
 			id4 := integration.NewIDAttribute("port", "1234")
 			pbEntity, err := testIntegration.Entity("testDB", "pgbouncer", id3, id4)
 			assert.Nil(t, err)
-			assert.Equal(t, expectedStats, pbEntity.Metrics[0].Metrics)
+			assert.Equal(t, len(pbEntity.Metrics), 2)
+			if testCase.expectedStats == nil {
+				assert.Equal(t, expectedStats, pbEntity.Metrics[0].Metrics)
+			} else {
+				assert.Equal(t, testCase.expectedStats, pbEntity.Metrics[0].Metrics)
+			}
```
Comment (sigilioso) [inline src/metrics/metrics_test.go:701]: The same for `expectedStats`

**Review by sigilioso — COMMENTED** [review]
Thanks for taking care of this! I've left some comments for us to discuss, let me know what you think.

By the way, we also need to to update the `CHANGELOG.md` file as in #176

File: src/metrics/metrics_test.go (hunk 1 above)
Comment (rahulreddy15) [reply src/metrics/metrics_test.go]: Makes sense. Will fix this.

File: src/metrics/metrics_test.go (hunk 2 above)
Comment (rahulreddy15) [reply src/metrics/metrics_test.go]: Fixed.

File: src/metrics/pgbouncer_definitions.go [hunk 3]
```diff
@@ -13,23 +13,25 @@ var pgbouncerStatsDefinition = &QueryDefinition{
...
+		AvgQueryCount              *int64 `db:"avg_query_count"   metric_name:"pgbouncer.stats.avgQueryCount"                                   source_type:"gauge"`
+		AvgServerAssignmentCount   *int64 `db:"avg_server_assignment_count" metric_name:"pgbouncer.stats.avgServerAssignmentCount" source_type:"gauge"`
+		AvgRecv                    *int64 `db:"avg_recv"          metric_name:"pgbouncer.stats.avgBytesIn"                                      source_type:"gauge"`
+		AvgSent                    *int64 `db:"avg_sent"          metric_name:"pgbouncer.stats.avgBytesOut"                                     source_type:"gauge"`
+		AvgReq                     *int64 `db:"avg_req"           metric_name:"pgbouncer.stats.avgRequestsPerSecond"                            source_type:"gauge"`
+		AvgQueryTime               *int64 `db:"avg_query_time"    metric_name:"pgbouncer.stats.avgQueryDurationInMilliseconds"                  source_type:"gauge"`
+		AvgQuery                   *int64 `db:"avg_query"         metric_name:"pgbouncer.stats.avgQueryDurationInMilliseconds"                  source_type:"gauge"`
+		AvgWaitTime                *int64 `db:"avg_wait_time"`
```
[the remark this answers is not in the document]
Comment (rahulreddy15) [reply src/metrics/pgbouncer_definitions.go]: Added comments.

File: tests/docker-compose-pgbouncer.yml
[the remark this answers is not in the document]
Comment (rahulreddy15) [reply tests/docker-compose-pgbouncer.yml:1]: tests/README.md now has steps to follow when adding support for new pgbouncer versions in the future.

**Review by sigilioso — APPROVED** [review]
LGTM 👍

**Pull request merged**